common/src/main/java/com/microsoft/identity/common/internal/telemetry/Telemetry.java

```diff
@@ -176,7 +173,8 @@ public synchronized void removeObserver(final ITelemetryObserver observer) {
      * @return List of ITelemetryObserver object.
      */
     public List<ITelemetryObserver> getObservers() {
-        return Collections.unmodifiableList(mObservers);
+        List observersList = new CopyOnWriteArrayList<>(mObservers);
+        return Collections.unmodifiableList(observersList);
```

Comment:
why a `CopyOnWriteArrayList` here , we are passing an unmodifiable list anyways right i.e. we should be thread safe already isn't it?

Reply:
This method returns a `List` and `Collections.unmodifiableList` accepts a list as the parameter, however `mObservers` is a queue instead. So first converting the `mObservers` queue to a list by using `CopyOnWriteArrayList` and then I create an Unmodifiable List by that.

Actually, I'm going to refactor this method to instead return a `queue` which is the actual data type for `mObservers` and I think that makes more sense, and I won't return an unmodifable version anymore as the queue will be thread-safe anyway.

Reply:
@kreedula I've pushed the change to return a Queue instead as discussed above. Let me know how it looks.

Reply:
@kreedula As discussed `mObservers` has been changed back to a `List`